File: backend/src/app/models/gpon/CtoCount.ts
import { BuildOptions, DataTypes, Model, Sequelize } from "sequelize";

export interface CtoCountAttributes {
    id: number;
    erp_ap_id: number;
    gpon_integration_id: number;
    description: string;
    online: number;
    offline: number;
}
export interface CtoCountModel extends Model<CtoCountAttributes>, CtoCountAttributes { }

export type CtoCountStatic = typeof Model & {
    new(values?: object, options?: BuildOptions): CtoCountModel;
};

export function CtoCountFactory(sequelize: Sequelize): CtoCountStatic {
    return <CtoCountStatic>sequelize.define("cto_count", {
        id: {
            type: DataTypes.INTEGER,
            autoIncrement: true,
            primaryKey: true,
        },
        erp_ap_id: {
            type: DataTypes.INTEGER,
        },
        gpon_integration_id: {
            type: DataTypes.INTEGER,
        },
        description: {
            type: DataTypes.STRING,
            allowNull: true,
        },
        online: {
            type: DataTypes.NUMBER,
        },
        offline: {
            type: DataTypes.NUMBER,
        },
    }, { tableName: "cto_count", timestamps: false });
}

File: snmpagent/app/models/gpon/CtoCount.py
from django.db import models
from app._models.bra_admin.Ap import Ap

class CtoCount(models.Model):
    description = models.CharField(max_length=100)
    online = models.IntegerField(null=False)
    offline = models.IntegerField(null=False)

    erp_ap_id = models.IntegerField()
    gpon_integration_id = models.IntegerField()

    class Meta:
        db_table = "cto_count"
    
    def __str__(self):
        return '{}|{}'.format(self.description, self.online)